pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -1383,6 +1383,10 @@ func (dm *DockerManager) KillPod(pod *api.Pod, runningPod kubecontainer.Pod, gra
 
 // NOTE(random-liu): The pod passed in could be *nil* when kubelet restarted.
 func (dm *DockerManager) killPodWithSyncResult(pod *api.Pod, runningPod kubecontainer.Pod, gracePeriodOverride *int64) (result kubecontainer.PodSyncResult) {
+	// Short circuit if there's nothing to kill.
+	if len(runningPod.Containers) == 0 {
```

Comment:
Yes, it does. (See line 1424 below). This method is a noop if `len(runningPod.Containers) == 0`, this is just an optimizitaion.